### PR TITLE
GEMM  fpga vectorized

### PR DESCRIPTION
**Pull request by TizianoDeMatteis**
New FPGA sample, with vectorization

### CODE REVIEW

File: samples/fpga/gemm_systolic_vectorized.py [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+import dace
+import numpy as np
+import pdb
+import select
+import sys
+
+N = dace.symbol("N")
+K = dace.symbol("K")
+M = dace.symbol("M")
+P = dace.symbol("P")
```
Comment (definelicht) [inline samples/fpga/gemm_systolic_vectorized.py:17]: You could also make `W` a symbol (we used to not support this, but it should work now)?

(if it doesn't work, forget about it)

File: samples/fpga/gemm_systolic_vectorized.py [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+    sdfg.add_stream("C_pipe",
+                    vec_type,
+                    transient=True,
+                    shape=(P + 1, ),
+                    storage=dace.dtypes.StorageType.FPGA_Local)
+
+    make_read_A(state)
+    make_read_B(state, sdfg, vec_width)
+    make_compute(sdfg, state, vec_width)
+    make_write_C(state, sdfg, vec_width)
```
Comment (definelicht) [inline samples/fpga/gemm_systolic_vectorized.py:350]: Are you sure you implemented `gemm`? Shouldn't there be a `read_C`?

**Review by definelicht — CHANGES_REQUESTED** [review]

File: samples/fpga/gemm_systolic_vectorized.py (hunk 2 above)
Comment (TizianoDeMatteis) [reply samples/fpga/gemm_systolic_vectorized.py:352]: it's in the writer

Comment (definelicht) [reply samples/fpga/gemm_systolic_vectorized.py:352]: Interesting idea :O Could you update the name?

**Review by definelicht — APPROVED** [review]

**Pull request merged**